foundry/hooks.d.ts

```diff
@@ -116,72 +139,255 @@ declare class Hooks {
+     * @param actor  - the Actor to whom this effect should be applied
+     * @param change - the change data being applied
+     * @see {@link ActiveEffect#_applyCustom}
+     */
+    applyActiveEffect: (actor: Actor, change: ActiveEffect.Change) => void;
+
+    /**
+     * This is called before a {@link Canvas} is drawn.
+     * @param canvas - the Canvas
+     * @see {@link Canvas#draw}
+     */
+    canvasInit: (canvas: Canvas) => void;
+
+    /**
+     * This is called when a {@link Canvas} is panned. When called during animated panning, the callback is called on
+     * every tick.
+     * @param canvas - the Canvas
+     * @param view   - the CanvasView
+     * @see {@link Canvas#pan}
+     * @see {@link Canvas#animatePan}
+     */
+    canvasPan: (canvas: Canvas, view: Canvas.View) => void;
+
+    /**
+     * This is called after a {@link Canvas} is done initializing.
+     * @param canvas - the Canvas
+     * @see {@link Canvas#draw}
+     */
+    canvasReady: (canvas: Canvas) => void;
+
+    /**
+     * This is called when creating a {@link ChatBubble}, but before displaying it.
+     * @param token   - the speaking token
+     * @param jq      - the JQuery for the chat bubble
+     * @param message - the spoken message text
+     * @param options - additional options
+     * @param emote   - whether to style the speech bubble as an emote
+     * @returns whether additional callbacks should be called after this
+     * @see {@link ChatBubbles#say}
+     */
+    chatBubble: (
+      token: Token,
+      jq: JQuery,
+      message: string,
+      options: {
+        emote: boolean;
+      }
+    ) => boolean;
+
+    /**
+     * This is called first when processing a chat message.
+     * @param chatLog  - the ChatLog
+     * @param message  - the original string of the message content
+     * @param chatData - the ChatData
+     * @returns whether additional callbacks should be called after this
+     * @see {@link ChatLog#processMessage}
+     */
+    chatMessage: (chatLog: ChatLog, message: string, chatData: ChatMessage.ChatData) => boolean;
+
+    /**
+     * This is called after the {@link SceneNavigation} is expanded or collapsed.
+     * @param nav       - the SceneNavigation
+     * @param collapsed - whether the navigation is collapsed
+     * @see {@link SceneNavigation#expand}
+     * @see {@link SceneNavigation#collapse}
+     */
+    collapseSceneNavigation: (nav: SceneNavigation, collapsed: boolean) => void;
+
+    /**
+     * This is called during the drop portion of a drag-and-drop event on an actor sheet.
+     * @param actor - the Actor the sheet belongs to
+     * @param sheet - the ActorSheet, the data was dropped on
+     * @param data  - the dropped data, already parsed as an object via JSON
+     * @returns whether additional callbacks should be called after this
+     * @see {@link ActorSheet#_onDrop}
+     */
+    dropActorSheetData: (actor: Actor, sheet: ActorSheet, data: object) => boolean;
```

Comment:
`data` should probably at least have the `type` property, maybe of type `Const.EntityTypes`.